data/csv.go

```diff
@@ -0,0 +1,32 @@
+package data
+
+import (
+	"encoding/csv"
+	"log"
+	"os"
+)
+
+type csvDataSource struct {
+	// Collection is a file path to a CSV file
+	Collection string
+}
+
+func NewCSVDataSource(csvPath string) csvDataSource {
```

Comment:
It's a best practice to add a descriptive comment above the functions, methods, type definitions, etc.
These comments work to help other developers to know what is this function/method/type about, and also have a purpose to work as documentation for your project.

You can extend as much or little as you want, but let's always add a descriptive comment.
```suggestion
// NewCSVDataSource function receives a cvsPath to initialize and return a csvDataSource instance
func NewCSVDataSource(csvPath string) csvDataSource {
```

Be careful, you are returning an unexported field. This might be annoying to use.